EpiAware/test/EpiLatentModels/models/RandomWalk.jl

```diff
@@ -38,3 +38,39 @@ end
     @test rw_process.init_prior == init_prior
     @test rw_process.std_prior == std_prior
 end
+
+@testitem "Testing RandomWalk parameter recovery: Negative Binomial errors on log rw" begin
+    using Random, Turing, FillArrays, Distributions, LinearAlgebra, DynamicPPL, StatsBase,
+          ReverseDiff
+    Random.seed!(1234)
+
+    rw_process = RandomWalk()
+    obs_nb = NegativeBinomialError()
+
+    @model function test_negbin_errors(rw, obs, y_t)
+        n = length(y_t)
+        @submodel Z_t = generate_latent(rw, n)
```

Comment:
you could save duplication by hard coding the rw and obs struct in the gen calls vs passing through

Reply:
Or loop over a few models to make this worthwhile?

Reply:
I've made a function `test_full_process` that ingests a latent model and a IGP model (the same matrix as the pipeline... not coincidently). The test matrix creates this array of models and runs `test_full_process` on each.